### ISSUE TITLE
Maybe false positive NP_NULL_ON_SOME_PATH (with Kotlin)

### ISSUE THREAD

**Issue by bgrozev** [spotbugs/spotbugs#1079]
I'm not sure whether kotlin is officially supported, but I ran `spotbugs` on a couple of projects and it produced useful results, so we'll continue to use it.

I found this code produces a `NP_NULL_ON_SOME_PATH` warning and I can't understand why:
```
class Test {
    val reason: String? by lazy { "non-null" }
}
```

With the type of `reason` changed to `String` no errors are produced. 

Here are the class files produced by the kotlin compiler:
https://drive.google.com/file/d/1TYnjC6qScR0WQR-eEPcirMmL1VjWpM3a/view?usp=sharing
https://drive.google.com/file/d/1X1D6L_sVLC3T6hX4Y81u7FuaVdva_pgG/view?usp=sharing

**Comment by sgflt** [comment]
Another case where it reports false positive NP_NONNULL_RETURN_VIOLATION on getter of lateinit field:

```kotlin
@Entity
@Table(name = "mytable")
internal class MyEntity {
    @Column(name = "mycolumn", nullable = false)
    @Convert(converter = MyCodeConverter::class)
    lateinit var myCode: MyCode
}
```
